src/cli/serve/serve.js

```diff
@@ -67,6 +69,7 @@ function readServerSettings(opts, extraCliOptions) {
     opts.pluginPath
   )));
 
+  merge(readKeystore());
   merge(extraCliOptions);
```

Comment:
do you think  it's worth adding a warning on conflicts or documented order of operations?  this isn't limited to this pr, so i'm cool with waiting

precedence looks roughly like
1) cli args
1) keystore
1) kibana.*yml
1) kibana.yml
1) defaults

Reply:
I agree both adding a warning, (or logging where a setting came from) and documenting the order would be helpful.

Reply:
https://github.com/elastic/kibana/issues/14903